### PR TITLE
Console ui

### PR DESCRIPTION
**Pull request by PashaBeshchuk**

### CODE REVIEW

**Comment by PashaBeshchuk** [comment]
> Will this work in browser?

Yes :)

**Comment by zbstof** [comment]
Ok, now add logging each change to console

File: tetris.js [hunk 1]
```diff
@@ -615,27 +615,48 @@ let tetris = {
...
+					if(field[y][x]){
+						fieldImage += `o`
+					}else if(this.coordinateMatchingCheck(tetromino, x, y)){
+						fieldImage += `X`
+					}else{
+						fieldImage += `-`
+					}	
+				}
+			}
+			if(y === fieldSize.y-1){
```
Comment (zbstof) [inline tetris.js]: why do you need this if?

File: tetris.js [hunk 2]
```diff
@@ -615,27 +615,48 @@ let tetris = {
...
-	// 			fieldImage = fieldImage + `\n            `
-	// 		}
+	},
+	displayField: function (fieldSize, tetromino, field){
+		let fieldImage;
+		for(let y = 0; y < fieldSize.y; y++){
+			for(let x = 0; x < fieldSize.x; x++){
+				if(!fieldImage){
+					if(field[y][x]){
+						fieldImage = "\n            "
```
Comment (zbstof) [inline tetris.js]: why do you need this huge space?
and don't say: so that tests pass

File: tetris.js [hunk 3]
```diff
@@ -615,27 +615,48 @@ let tetris = {
...
+						fieldImage += `o`
+					}else if(this.coordinateMatchingCheck(tetromino, x, y)){
+						fieldImage = "\n            "
+						fieldImage += `X`
+					}else{
+						fieldImage = "\n            "
```
Comment (zbstof) [inline tetris.js]: you always add newline in this if branch, move outside inner ifs

File: tetris.js [hunk 4]
```diff
@@ -615,27 +615,48 @@ let tetris = {
...
+						fieldImage += `-`
+					}
+				}else {
```
Comment (zbstof) [inline tetris.js]: this else branch can be merged with the branch in if

File: tetris.js [hunk 5]
```diff
@@ -615,27 +615,48 @@ let tetris = {
...
+					if(field[y][x]){
```
Comment (zbstof) [inline tetris.js]: formatting

File: tetris.js (hunk 3 above)
Comment (PashaBeshchuk) [reply tetris.js]: Тут вроде бы верно

File: tetris.js (hunk 2 above)
Comment (PashaBeshchuk) [reply tetris.js]: во первых для красоты в консоле
Во вторых у меня тесты так написаны, я могу убрать отступы, но тест будет не красивый

File: tetris.js (hunk 4 above)
Comment (PashaBeshchuk) [reply tetris.js]: Не понимаю о чем ты:(

File: tetris.js [hunk 6]
```diff
@@ -616,5 +615,48 @@ module.exports = {
 		rotate() {
 			this.tetromino.rotateTetromino(this.field.field)
 		}
+	},
+	displayField: function (fieldSize, tetromino, field){
```
Comment (zbstof) [inline tetris.js]: why do you need to pass fieldSize? it is a part of field

File: tetris.js (hunk 2 above)
Comment (zbstof) [reply tetris.js]: do it anyway

File: tetris.js (hunk 3 above)
Comment (zbstof) [reply tetris.js]: do it anyway

File: tetris.js (hunk 2 above)
Comment (zbstof) [reply tetris.js]: just initialize fieldImage in line 620, and you won't need if in line 623

File: tetris.js [hunk 7]
```diff
@@ -616,43 +616,28 @@ let tetris = {
...
-					}else if(this.coordinateMatchingCheck(tetromino, x, y)){
-						fieldImage += `X`
-					}else{
-						fieldImage += `-`
-					}	
+	displayField: function (fieldSize, tetromino, field) {
+		let fieldImage = "\n            ";
+		for (let y = 0; y < fieldSize.y; y++) {
+			if(y > 0){
+				fieldImage += "\n            ";
```
Comment (zbstof) [inline tetris.js]: just add newline at the end of the loop iteration, not at the beginning

File: tetris.js [hunk 8]
```diff
@@ -616,43 +616,28 @@ let tetris = {
...
 				}
 			}
-			if(y === fieldSize.y-1){
-				fieldImage = fieldImage + `\n        `
-			}else{
-				fieldImage = fieldImage + `\n            `
-			}
-			
 		}
+		fieldImage += "\n        ";
```
Comment (zbstof) [inline tetris.js]: ... then you won't need this crutch

File: tetris.js (hunk 6 above)
Comment (zbstof) [reply tetris.js]: this is not fixed

**Pull request merged**